### DISCUSSION TITLE
How to get HTTP headers?

### DISCUSSION THREAD

**Discussion by tuanlt2** [graphql-java-kickstart/graphql-java-tools#475] [Q&A]
How to get HTTP headers?

**Comment by lmeadors** [comment]
I'm using an `ExecutionInputCustomizer` - in it's `customizeExecutionInput` method, you have access to both the `WebRequest` object and the `executionInput` objects - so you can copy headers from the incoming web request to the current context.

**Reply by KushagraBindal** [reply]
@lmeadors Larry I am also having the similar requirement where I need to extract the HttpHeaders Object from the request. Can you please help here with some example that how did you achieve the same inside 

public DataResponse get(DataFetchingEnvironment environment) {}

**Reply by oliemansm** [reply]
```java
public DataResponse get(DataFetchingEnvironment environment) {
  GraphQLServletContext context = environment.getContext();
  HttpServletRequest request = context.getHttpServletRequest();
}
```
